src/java/com/puppetlabs/puppetserver/ShellUtils.java

```diff
@@ -43,13 +50,14 @@ private static ExecutionResult executeExecutor(CommandLine commandLine,
 
         Integer exitCode = executor.execute(commandLine, env);
 
+        ByteArrayInputStream stdoutInputStream = new ByteArrayInputStream(stdoutOutputStream.toByteArray());
         String stdErr = errStream.toString();
 
         if ( ! stdErr.isEmpty() ) {
             log.warn("Executed an external process which logged to STDERR: " + stdErr);
         }
 
-        return new ExecutionResult(outStream.toString(), stdErr, exitCode);
+        return new ExecutionResult(stdoutInputStream, stdErr, exitCode);
```

Comment:
I'm not sure what the standard practice is here, but you could inline `new ByteArrayInputStream(stdoutOutputStream.toByteArray())` here and not have the assignment above, as all it is used for is the return.